src/Shared/src/CommandLineUtils/CommandLine/CommandLineApplication.cs

```diff
@@ -153,24 +163,40 @@ public int Execute(params string[] args)
 
                         if (option == null)
                         {
-                            if (string.IsNullOrEmpty(longOptionName) && !command._throwOnUnexpectedArg  && AllowArgumentSeparator)
+                            var ignoreContinueAfterUnexpectedArg = false;
+                            if (string.IsNullOrEmpty(longOptionName) &&
+                                !command._throwOnUnexpectedArg &&
+                                AllowArgumentSeparator)
                             {
-                                // skip over the '--' argument separator
+                                // Skip over the '--' argument separator then consume all remaining arguments. All
+                                // remaining arguments are unconditionally stored for further use.
                                 index++;
+                                ignoreContinueAfterUnexpectedArg = true;
+                            }
+
+                            if (HandleUnexpectedArg(
+                                command,
+                                args,
+                                index,
+                                argTypeName: "option",
+                                ignoreContinueAfterUnexpectedArg))
+                            {
+                                continue;
                             }
 
-                            HandleUnexpectedArg(command, args, index, argTypeName: "option");
                             break;
                         }
 
                         // If we find a help/version option, show information and stop parsing
                         if (command.OptionHelp == option)
                         {
+                            arguments?.Dispose();
```

Comment:
Is this required for 3.0? I'm worried a change like this in the release late might be destabilizing.

Reply:
This was a VS suggestion I took because it made sense. `arguments` is a local variable (see https://github.com/aspnet/Extensions/pull/2210/files#diff-b972a436637005ce5a174421167fd548R138) referencing a disposable object. My preference is not to leave the VS message glowing in 3.0.

Also, what could this destabilize?

Reply:
Let's remove this if it's not absolutely required. We don't have a way to test the impact of this change in all the places this gets used and it doesn't seem required for service reference to work.